src/components/Drawer/PDrawer.vue

```diff
@@ -0,0 +1,175 @@
+<template>
+  <teleport to="body">
+    <transition name="p-drawer__slide" :duration="350">
+      <keep-alive>
+        <PLayoutResizable v-if="modelValue" :disabled="!resizable" class="p-drawer" :class="classes.root" :placement="placement">
+          <template #aside>
+            <div class="p-drawer__aside" :style="styles" :class="attrClasses" v-bind="{ ...listeners, ...attrs }">
+              <slot v-bind="{ show, hide, toggle }" />
+            </div>
+          </template>
+
+          <slot name="overlay" v-bind="{ show, hide, toggle }">
+            <div class="p-drawer__overlay" @click="toggle" />
+          </slot>
+        </PLayoutResizable>
+      </keep-alive>
+    </transition>
+  </teleport>
+</template>
+
+<script lang="ts">
+  export default {
+    name: 'PDrawer',
+    expose: [],
+    components: { PLayoutResizable },
+    inheritAttrs: false,
+  }
+</script>
+
+<script lang="ts" setup>
+  import { computed, ref } from 'vue'
+  import { useAttrsStylesClassesAndListeners } from '@/compositions'
+  import { PLayoutResizable } from '@/layouts'
+
+  export type PDrawerPlacement = 'left' | 'right' | 'top' | 'bottom'
+
+  const props = defineProps<{
+    modelValue?: boolean,
+    resizable?: boolean,
+    placement?: PDrawerPlacement,
+  }>()
+
+  const emit = defineEmits<{
+    (event: 'update:modelValue', value: boolean): void,
+  }>()
+
+  const placement = computed(() => props.placement ?? 'left')
+
+  const { classes: attrClasses, listeners, styles, attrs } = useAttrsStylesClassesAndListeners()
+
+  const internalValue = ref(props.modelValue)
+  const modelValue = computed<boolean>({
+    get() {
+      return props.modelValue || internalValue.value
+    },
+    set(value) {
+      internalValue.value = value
+      emit('update:modelValue', value)
+    },
+  })
+
+  const hide = (): void => {
+    modelValue.value = false
+  }
+
+  const toggle = (): void => {
+    modelValue.value = !modelValue.value
+  }
+
+  const show = (): void => {
+    modelValue.value = true
+  }
```

Comment:
This is identical to the `useShowModal` in ui library. Maybe we should rename that to `useShow` and move it into prefect-design.

Reply:
We could even make a generic `useShow` composition and then re-export it as `useShowDrawer` and `useShowModal` for a nice devx.